classes/Entities/HelpdeskProduct.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Avado\MoodleAbstractionLibrary\Entities;
+
+/**
+ * Class HelpdeskProduct
+ * @package Avado\MoodleAbstractionLibrary\Entities
+ */
+class HelpdeskProduct extends BaseModel
+{
+    /**
+     * @var string
+     */
+    protected $table = 'helpdesk_products';
+
+    /**
+     * Get the course_categories id, via the helpdesk_product_courses table
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasManyThrough
+     */
+    public function productCategories(){
+        return $this->hasManyThrough(HelpdeskProduct::class, HelpdeskProductCourse::class, 'coursecategoryid', 'id', 'id', 'id');
```

Comment:
**hasManyThrough** used when we have 3 tables and Table 1 connected to Table 2, Table 2 connected to Table 3 then we can establish connection b/w Table 1 and Table 3 using **hasManyThrough**. 
For your current need I guess **hasMany** will work.

Reply:
Yes I have connected table 1 (helpdesk_product) to table 3 (course_categories), via table 2 (helpdesk_product_courses). Because I wanted to be able to have the course_categories id for HelpdeskProduct (needed for the initial ticket), which does not have access to it, but have access to helpdesk_product_course table.

But yes you are right, I can connect directly helpdesk_product_courses and course_categories with hasMany.
I will change it.
In that case, if I am right I should also change the place of this function which should be in HelpdeskProductCourse model instead of HelpdeskProduct.

Reply:
If HelpdeskProduct can have multiple Courses then this is correct if vice-versa then you can change place.